overlord/ifacestate/handlers.go

```diff
@@ -314,7 +315,33 @@ func (m *InterfaceManager) setupProfilesForAppSet(task *state.Task, appSet *inte
 			return err
 		}
 
-		appSet, err := appSetForSnapRevision(st, snapInfo)
+		var appSet *interfaces.SnapAppSet
+		if snapst.PendingSecurity != nil {
+			// a content plug/slot may have already updated in this change, so the appSet
+			// should reflect in the revision (otherwise, we may regenerate the
+			// profile for the wrong revision)
+			snapInfo.SideInfo = *snapst.PendingSecurity.SideInfo
+
+			var comps []*snap.ComponentInfo
+			for _, compSi := range snapst.PendingSecurity.Components {
```

Comment:
We have similar code in `SnapState.ComponentInfosForRevision` - would it make sense to move this code to its own function in *SnapState with a different API?

Reply:
Sure, I've added a new helper. Thanks